js/components/grid/calendar.js.jsx

```diff
@@ -99,7 +114,11 @@ class TimetableHeader extends React.Component {
         <thead>
           <tr>
             {dayCells}
-            <th className="term-name">Spring</th>
+            <th className="term-name">
+              Spring
+              <MapModal ref={ r => this.modal = r}/>
```

Comment:
Don't duplicate the `MapModal`. Instead, create just one, and eventually pass in as props the necessary data. Right now, you don't need any props because there's no real data, but you could change the `displayMap` method to take an a semester parameter (e.g., `() => this.displayMap('F')`).